File: lib/domain/cultivation_plan/dtos/cultivation_plan_destroy_output.rb
# frozen_string_literal: true

module Domain
  module CultivationPlan
    module Dtos
      class CultivationPlanDestroyOutput
        attr_reader :undo

        def initialize(undo:)
          @undo = undo
        end
      end
    end
  end
end

File: crates/agrr-domain/src/cultivation_plan/dtos/cultivation_plan_destroy_output.rs
//! Ruby: `Domain::CultivationPlan::Dtos::CultivationPlanDestroyOutput`

use serde_json::Value;

#[derive(Debug, Clone, PartialEq)]
pub struct CultivationPlanDestroyOutput {
    pub undo: Value,
}

impl CultivationPlanDestroyOutput {
    pub fn new(undo: Value) -> Self {
        Self { undo }
    }
}
